providers/amazon/src/airflow/providers/amazon/aws/triggers/mwaa.py

```diff
@@ -87,14 +87,95 @@ def __init__(
             return_value=external_dag_run_id,
             waiter_delay=waiter_delay,
             waiter_max_attempts=waiter_max_attempts,
-            aws_conn_id=aws_conn_id,
             waiter_config_overrides={
                 "acceptors": _build_waiter_acceptors(
                     success_states=self.success_states,
                     failure_states=self.failure_states,
                     in_progress_states=in_progress_states,
                 )
             },
+            **kwargs,
+        )
+
+    def hook(self) -> AwsGenericHook:
+        return MwaaHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region_name,
+            verify=self.verify,
+            config=self.botocore_config,
+        )
+
+
+class MwaaTaskCompletedTrigger(AwsBaseWaiterTrigger):
+    """
+    Trigger when an MWAA Task is complete.
+
+    :param external_env_name: The external MWAA environment name that contains the DAG Run you want to wait for
```

Comment:
```suggestion
    :param external_env_name: The external MWAA environment name that contains the Task Instance you want to wait for
```

Same as the Operator class, these param descriptions need slight updates for task waiting not dag run waiting.